### PR TITLE
fix!: avoid float calculations in PlatformShare

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
avoid potential discrepancies in block reward calculations

## What was done?
use integers (int64_t) only when dealing with block rewards, no float/double

## How Has This Been Tested?
run tests

## Breaking Changes
might fork off on devnets that use previous version

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by ogabrielides — APPROVED** [review]
ACK

File: test/functional/feature_asset_locks.py [hunk 1]
```diff
@@ -515,7 +515,7 @@ def run_test(self):
         all_mn_rewards = platform_reward + owner_reward + operator_reward
         all_mn_rewards += 1 * 0.75
         assert_equal(all_mn_rewards, bt['coinbasevalue'] * 0.75)  # 75/25 mn/miner reward split
-        assert_equal(platform_reward, int(all_mn_rewards * 0.375))  # 0.375 platform share
+        assert_equal(platform_reward, int(all_mn_rewards * 375 // 1000))  # 0.375 platform share
```
[the remark this answers is not in the document]
Comment (knst) [reply test/functional/feature_asset_locks.py:518]: // is an integer division, so, behaviour would be same as in C++ implementation

[the remark this answers is not in the document]
Comment (UdjinM6) [reply test/functional/feature_asset_locks.py:518]:
```suggestion
        assert_equal(platform_reward, all_mn_rewards * 375 // 1000)  # 0.375 platform share
```

**Review by knst — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**